gtwrap/interface_parser/function.py

```diff
@@ -29,15 +29,29 @@ class Argument:
     void sayHello(/*`s` is the method argument with type `const string&`*/ const string& s);
     ```
     """
-    rule = ((Type.rule ^ TemplatedType.rule)("ctype") +
-            IDENT("name")).setParseAction(lambda t: Argument(t.ctype, t.name))
+    rule = ((Type.rule ^ TemplatedType.rule)("ctype") + IDENT("name") +
+            Optional(EQUAL + (DEFAULT_ARG ^ Type.rule ^ TemplatedType.rule) +
```

Comment:
Should this maybe be
```python
Optional( EQUAL +
  (DEFAULT_ARG ^
    ( (Type.rule ^ TemplatedType.rule) + Optional(LPAREN + RPAREN) )
  )
)
```
???

i.e. It can _either_ be a default arg, or a constructor?

Also maybe `Optional(LPAREN + DEFAULT_ARG + RPAREN)` ?  so that the constructor can take constant arguments?

Actually there could even be nested constructors... oh jeez this is getting really involved

Reply:
The rule handles the left and right parens properly, check out the tests. I've added a check where if it is a Type, then to add back the parens.

My design sense says not to support non-default constructors. If the user wanted non-default constructors, then why use a default argument in the first place?

Reply:
That being said, maybe we can replace Type/TemplatedType with a Constructor rule?

Reply:
If the constructor rule supports templating*